autoload/phpactor/nvim.vim

```diff
@@ -0,0 +1,49 @@
+let s:lock = v:null
+
+func! phpactor#nvim#asyncCall(action, arguments)
+    " TODO: Include either the original "action" in the response or a request
+    " ID (probably better to include both)
+    if s:lock != v:null
+        echo "An asynchronous RPC '" . s:lock . "' action is already running"
+        return
+    endif
+
+    let s:lock = a:action
+    let callbacks = {
+    \   'on_stdout': function('phpactor#nvim#asyncHandle'),
+    \   'on_stderr': function('phpactor#nvim#asyncHandle'),
+    \   'on_exit': function('phpactor#nvim#asyncHandle')
+    \ }
+    let job = jobstart([ g:phpactorPhpBin, g:phpactorbinpath, 'rpc', '--working-dir=' . g:phpactorInitialCwd ], callbacks)
+
+    let request = { "action": a:action, "parameters": a:arguments }
+
+    call chansend(job, json_encode(request))
+    call chanclose(job, 'stdin')
+
+    return job
+endfunc
+
+let s:stdout = []
```

Comment:
I also still have no idea how to catch the response without using non-local scope (like the above). 

What do you think about creating a RPC server between vim and one-time phpactor commands? PHPCD mentioned already somewhere had [a such solution](https://github.com/przepompownia/phpcd.vim/blob/master/autoload/rpc.vim). 

I did not tested behaviour of vim-specific `ch_evalraw`.

Reply:
One idea of this PR was to allow messages to be sent _from_ Phpactor during a command's execution (i.e. to send progress notifications from the Indexer).

I'm not sure there are benefits to having an 2-way RPC connection - I can only imagine the use-case would be to cancel an operation - and in that case sending a `SIGTERM` from VIM should be enough?

Reply:
But in anycase it's worth concentrating on the language server solution for now - we can support most of Phpactor's operations (even refactoring) over LSP and it if it works well, it could replace the Phpactor VIM client one day.